src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1522,7 +1523,9 @@ internal static bool ReportInvalidScopedOverrideAsError(MethodSymbol baseMethod,
             {
                 var baseParameter = baseParameters[i];
                 var overrideParameter = overrideParameters[i + overrideParameterOffset];
-                if ((baseParameter.RefKind, overrideParameter.RefKind) is (RefKind.RefReadOnlyParameter, RefKind.In) or (RefKind.In, RefKind.RefReadOnlyParameter))
+                if (methodGroupConversion
+                    ? baseParameter.RefKind != overrideParameter.RefKind
+                    : (baseParameter.RefKind, overrideParameter.RefKind) is (RefKind.RefReadOnlyParameter, RefKind.In) or (RefKind.In, RefKind.RefReadOnlyParameter))
```

Comment:
>(baseParameter.RefKind, overrideParameter.RefKind) is (RefKind.RefReadOnlyParameter, RefKind.In) or (RefKind.In, RefKind.RefReadOnlyParameter) [](http://example.com/codeflow?start=22&length=142)

Do we need this detailed check? Could we always do a simple check ```baseParameter.RefKind != overrideParameter.RefKind```?

Reply:
Right, we can since we get here only for valid OHI mismatches anyway.

Reply:
Probably can get rid of the ```methodGroupConversion``` parameter too.